.github/workflows/release-next.yml

```diff
@@ -0,0 +1,54 @@
+name: Release Next
+
+on:
+  push:
+    branches:
+      - main
+
+env:
+  NODE_VERSION: 18.10.0
+  PNPM_VERSION: 8
+  NX_CLOUD_ACCESS_TOKEN: ${{ secrets.NX_CLOUD_ACCESS_TOKEN }}
+
+jobs:
+  release:
+    name: Release Next
+    runs-on: ubuntu-latest
+    steps:
+      - name: Get token
+        id: get_token
+        uses: tibdex/github-app-token@v2
+        with:
+          private_key: ${{ secrets.RELEASE_APP_PEM }}
+          app_id: ${{ secrets.RELEASE_APP_ID }}
+      - name: Checkout code
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+          # persist-credentials: false
+          token: ${{ steps.get_token.outputs.token }}
+          ref: ${{ github.ref }}
+      - run: |
+          git config user.name github-actions
+          git config user.email github-actions@github.com
+      - uses: actions/setup-node@v4
+        with:
+          node-version: ${{ env.NODE_VERSION }}
+          registry-url: https://registry.npmjs.org/
+      - uses: pnpm/action-setup@v4
+        with:
+          version: ${{ env.PNPM_VERSION }}
+      - name: Install dependencies
+        run: pnpm install --frozen-lockfile --ignore-scripts
+        env:
+          CI: true
+      - name: Build
+        run: npm run version:ci
+        env:
+          CI: true
+          GITHUB_TOKEN: ${{ steps.get_token.outputs.token }}
+      - name: Run pnpm publish
+        run: pnpm -r publish --access=public --no-git-checks --tag next
```

Comment:
it will publish all packages, not just react
In addition, I think it's not running on affected packages only, but on all packages
Another thing is the version numbers, I'm not sure if it will generate the same version pattern we have for our next versions

Did you give it a try?